android/src-huawei/com/badlogic/gdx/pay/app/AndroidLauncher.java

```diff
@@ -0,0 +1,118 @@
+package com.badlogic.gdx.pay.app;
+
+import android.content.Intent;
+import android.content.IntentSender;
+import android.util.Log;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.pay.Transaction;
+import com.badlogic.gdx.pay.android.huawei.HuaweiPurchaseManager;
+import com.badlogic.gdx.pay.android.huawei.IAPListener;
+import com.huawei.hms.iap.Iap;
+import com.huawei.hms.iap.IapApiException;
+import com.huawei.hms.iap.IapClient;
+import com.huawei.hms.iap.entity.ConsumeOwnedPurchaseResult;
+import com.huawei.hms.iap.entity.InAppPurchaseData;
+import com.huawei.hms.iap.entity.OrderStatusCode;
+import com.huawei.hms.iap.entity.PurchaseIntentResult;
+import com.huawei.hms.iap.entity.PurchaseResultInfo;
+
+import org.json.JSONException;
+
+/**
+ * Created by Francesco Stranieri on 11.05.2020.
+ */
+
+public class AndroidLauncher extends GenericAndroidLauncher implements IAPListener {
```

Comment:
It shouldn't be necessary to make the Launcher class implement IAPListener interface. All Huawei-IAP-related stuff should be done by HuaweiPurchaseManager. Callbacks with IAP related stuff should go to gdx-pay's `PurchaseObserver`. I can't see where PurchaseObserver.handlePurchase is called here and I don't understand why you need an error handling here while HuaweiPurchaseManager also has some error handling calling PurchaseObserver's error methods.